api-put-object-progress.go

```diff
@@ -0,0 +1,122 @@
+/*
+ * Minio Go Library for Amazon S3 Compatible Cloud Storage (C) 2015 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package minio
+
+import (
+	"io"
+	"sync"
+)
+
+var blackHolePool = sync.Pool{
+	New: func() interface{} {
+		b := make([]byte, 8*1024*1024)
+		return b
+	},
+}
+
+func ioProgressN(writer io.Writer, reader io.Reader, size int64) (written int64, err error) {
```

Comment:
This function doesnt make sense. It should be replaced with:  io.CopyN(ioutil.Discard, source, size).